engine/grammar/matchers/datatypes/semver.go

```diff
@@ -0,0 +1,202 @@
+package datatypes
+
+import (
+	"fmt"
+	"strconv"
+
+	"strings"
+)
+
+const (
+	MetadataDelimiter   = "+"
```

Comment:
are these constants required to be public?

Reply:
updated